### PR TITLE
[Mellanox] Refactor SFP to use new APIs.

### PR DESCRIPTION
**Pull request by oleksandrivantsiv**
#### Why I did it
Refactor SFP code to remove code duplication and to be able to use the latest features available in new APIs.

#### How I did it
Refactor SFP code to remove code duplication and to be able to use the latest features available in new APIs.

#### How to verify it
Run sonic-mgmt/platform_tests/sfp tests

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111

### CODE REVIEW

File: platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py [hunk 1]
```diff
@@ -339,28 +339,13 @@ def get_change_event(self, timeout=0):
...
         if status:
-            self.reinit_sfps(port_dict)
             result_dict = {'sfp':port_dict}
             if error_dict:
                 result_dict['sfp_error'] = error_dict
             return True, result_dict
         else:
             return True, {'sfp':{}}
 
-    def reinit_sfps(self, port_dict):
```
Comment (keboliu) [inline platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py]: this function was added to address an issue in a corner case: replace a QSFP module with an SFP module(with a QSA adaptor) in the fly, w/o reinit, sfp object will not update the module type thus not able to identify the new type of module correctly, not sure whether the new implementation can overcome this case w/o this function.

Comment (oleksandrivantsiv) [reply platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py]: Thanks for letting me know. Now I understand the purpose of reinit method. Updated the implementation accordingly.

**Comment by oleksandrivantsiv** [comment]
@keboliu @liat-grozovik please review and provide your feedback

**Comment by liat-grozovik** [comment]
LGTM but the expert is Kebo. if he approves i am OK.
please sync also with Roy how to qualify it before publishing PR against Azure

**Review by keboliu — APPROVED** [review]

**Comment by liat-grozovik** [comment]
great. so next
1. run CI
2. run build, provide to verification and ask to run platform tests to have it approved for public PR.

**Pull request merged**